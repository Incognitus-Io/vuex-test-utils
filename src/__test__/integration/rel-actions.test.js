describe('Release Library', () => {
    describe('Asynchronous action', () => {
        let actions;

        beforeEach(() => {
            actions = {
                async foobar(_, __) { return; },
            };
        });

        describe('commit', () => {
            it('Should commit a mutation', () => {
                actions.foobar = async (ctx) => {
                    ctx.commit('fizzbuzz');
                };

                expect.action(actions.foobar)
                    .toCommit('fizzbuzz');
            });

            it('Should fail when commit types do not match', () => {
                const expectedCommit = 'somthing not right';
                actions.foobar = async (ctx) => {
                    ctx.commit('foobar');
                };

                try {
                    expect.action(actions.foobar)
                        .toCommit(expectedCommit);
                    assert.fail();
                } catch (err) {
                    expect(err.message).toMatchSnapshot();
                }
            });
        });

        describe('dispatch', () => {
            it('Should fail when dispatch types do not match', () => {
                const expectedDispatch = 'somthing not right';
                actions.foobar = async (ctx) => {
                    ctx.dispatch('foobar');
                };

                try {
                    expect.action(actions.foobar)
                        .toDispatch(expectedDispatch);
                    assert.fail();
                } catch (err) {
                    expect(err.message).toMatchSnapshot();
                }
            });
        });
    });
});
