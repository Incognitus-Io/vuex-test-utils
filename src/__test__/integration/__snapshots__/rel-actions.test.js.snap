// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release Library Asynchronous action commit Should fail when commit types do not match 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toCommit()[22m

Expected to have been commited:
 [32m\\"somthing not right\\"[39m
 but received:
 [31m[\\"foobar\\"][39m"
`;

exports[`Release Library Asynchronous action dispatch Should fail when dispatch types do not match 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toDispatch()[22m

Expected to have been dispatched:
 [32m\\"somthing not right\\"[39m
 but received:
 [31m[\\"foobar\\"][39m"
`;
